documentapi/src/main/java/com/yahoo/documentapi/messagebus/protocol/AdaptiveLoadBalancer.java

```diff
@@ -0,0 +1,55 @@
+// Copyright Verizon Media. Licensed under the terms of the Apache 2.0 license. See LICENSE in the project root.
+package com.yahoo.documentapi.messagebus.protocol;
+
+import com.yahoo.jrt.slobrok.api.Mirror;
+
+import java.util.List;
+import java.util.Random;
+
+/**
+ * Will pick 2 random candidates and select the one with least pending operations.
+ *
+ * @author baldersheim
+ */
+class AdaptiveLoadBalancer extends LoadBalancer {
+    private Random random = new Random();
+    AdaptiveLoadBalancer(String cluster) {
+        super(cluster);
+    }
+
+    @Override
+    Node getRecipient(List<Mirror.Entry> choices) {
+        if (choices.isEmpty()) return null;
+        Mirror.Entry entry;
+        NodeMetrics metrics;
+        if (choices.size() == 1) {
+            entry = choices.get(0);
+            metrics = getNodeMetrics(entry);
+        } else {
+            int candidateA = 0;
+            int candidateB = 1;
+            if (choices.size() > 2) {
+                candidateA = random.nextInt(choices.size());
+                candidateB = random.nextInt(choices.size());
+            }
```

Comment:
You tell me, will it have any statistical impact, or will it just drown in randomness.

Reply:
Probably the latter, although I'm not smart enough to crunch the numbers on it 🙂 Easiest would be to measure it in practice in the unit test over a set of runs. If there is a measurable better distribution if the two candidates are forced to be unequal, it would make sense to include it.

Reply:
After a reading a bit on the algorithm I concluded that it would indeed be better to ensure 2 distinct candidates. The reason is that when same candidate is selected fro both A and B it is in-fact a degradation into pure random. Looking at the numbers made it a simple choice.